Source/Charts/Data/Implementations/Standard/ChartDataSet.swift

```diff
@@ -517,3 +490,82 @@ open class ChartDataSet: ChartBaseDataSet
         return copy
     }
 }
+
+// MARK: MutableCollection
+extension ChartDataSet: MutableCollection {
+    public typealias Index = Int
+    public typealias Element = ChartDataEntry
+
+    public var startIndex: Index {
+        return values.startIndex
+    }
+
+    public var endIndex: Index {
+        return values.endIndex
+    }
+
+    public func index(after: Index) -> Index {
+        return values.index(after: after)
+    }
+
+    public subscript(position: Index) -> Element {
+        get {
+            // This is intentionally not a safe subscript to mirror
+            // the behaviour of the built in Swift Collection Types
+            return values[position]
+        }
+        set {
+            isIndirectValuesCall = true
```

Comment:
in `entryForIndex()` there is no `isIndirectValuesCall` and `calcMinMax`, why we need it here?
I'm a little concerned about `isIndirectValuesCall` here, forgetting to set it would be a trouble.

Reply:
`entryForIndex(_:)` is a get only method, note that the getter doesn't call `calcMinMax`. The setter allows arbitrary insertion of another element. If we insert an item we need to `calcMinMax`

Reply:
so this is a new implementation right? do you remember what's the old function equivalent to this setter?

Reply:
We never had one. It would have been something like `insert(element:atIndex:)`